x-pack/plugins/apm/public/hooks/use_fallback_to_transactions_fetcher.tsx

```diff
@@ -14,17 +15,31 @@ export function useFallbackToTransactionsFetcher({ kuery }: { kuery: string }) {
   const rangeFrom = 'rangeFrom' in query ? query.rangeFrom : undefined;
   const rangeTo = 'rangeTo' in query ? query.rangeTo : undefined;
 
+  const device = 'device' in query ? query.device : undefined;
+  const osVersion = 'osVersion' in query ? query.osVersion : undefined;
+  const appVersion = 'appVersion' in query ? query.appVersion : undefined;
+  const netConnectionType =
+    'netConnectionType' in query ? query.netConnectionType : undefined;
+
+  const kueryWithFilters = getKueryWithMobileFilters({
+    device,
```

Comment:
At the moment, I couldn't think of a better approach. However, here is the only place we have mobile logic.

cc @sqren